### ISSUE TITLE
Metadata is not allowed to be a list

### ISSUE THREAD

**Issue by rix0rrr** [aws-cloudformation/cfn-lint#3535]
### CloudFormation Lint Version

1.8.2

### What operating system are you using?

Mac

### Describe the bug

I get an error if I have a template that has an array value in Metadata:

```
Resources:
  MyBucket:
    Type: AWS::S3::Bucket
    Metadata:
      TakesAList:
        - a
        - b
        - c
```

```
$ cfn-lint bla.yaml
E3028 ['a', 'b', 'c'] is not of type 'object', 'string'
bla.yaml:5:7
```

### Expected behavior

A list is a valid data type, so I feel it should be allowed.

The documentation doesn't say it's not allowed: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-metadata.html

In fact it states that CloudFormation doesn't do anything with the `Metadata` key, so it shouldn't matter what it is.

### Reproduction template

```yaml
Resources:
  MyBucket:
    Type: AWS::S3::Bucket
    Metadata:
      TakesAList:
        - a
        - b
        - c
```
